sessionctx/variable/sysvar.go

```diff
@@ -1185,9 +1185,10 @@ var defaultSysVars = []*SysVar{
 		return config.GetGlobalConfig().Plugin.Dir, nil
 	}},
 	{Scope: ScopeGlobal, Name: TiDBEnableHistoricalStats, Value: Off, Type: TypeBool, GetGlobal: func(s *SessionVars) (string, error) {
-		return getTiDBTableValue(s, "tidb_enable_historical_stats", Off)
+		return s.GlobalVarsAccessor.GetGlobalSysVar(TiDBEnableHistoricalStats)
 	}, SetGlobal: func(s *SessionVars, val string) error {
-		return setTiDBTableValue(s, "tidb_enable_historical_stats", val, "Current historical statistics enable status")
+		SetSysVar(TiDBEnableHistoricalStats, val)
+		return nil
 	}},
```

Comment:
You can just remove the GetGlobal and SetGlobal funcs. This will be done automatically.